projects/kit/components/input-month-range/input-month-range.directive.ts

```diff
@@ -10,16 +20,53 @@ import {TuiInputMonthRangeComponent} from './input-month-range.component';
             provide: TUI_TEXTFIELD_HOST,
             useExisting: forwardRef(() => TuiInputMonthRangeDirective),
         },
+        TuiDestroyService,
     ],
 })
-export class TuiInputMonthRangeDirective extends TuiAbstractTextfieldHost<TuiInputMonthRangeComponent> {
+export class TuiInputMonthRangeDirective
+    extends TuiAbstractTextfieldHost<TuiInputMonthRangeComponent>
+    implements DoCheck
+{
+    private readonly value$ = new Subject<TuiMonthRange | null>();
+
+    private localizedValue: [string, string] = ['', ''];
+
+    constructor(
+        @Inject(AbstractTuiControl) host: TuiInputMonthRangeComponent,
+        @Inject(TUI_MONTH_FORMATTER)
+        formatter: TuiHandler<TuiMonth | null, Observable<string>>,
+        @Self() @Inject(TuiDestroyService) destroy$: Observable<unknown>,
+    ) {
+        super(host);
+
+        this.value$
+            .pipe(
+                distinctUntilChanged(),
+                switchMap((value: TuiMonthRange | null) =>
+                    combineLatest([
+                        formatter(value && value.from),
+                        formatter(value && value.to),
```

Comment:
`formatter(value?.from)`
`formatter(value?.to)`

Reply:
why not `TuiHandler<TuiMonth | null | undefined, ...`?)

Reply:
Because it is meant to work with a form control and lack of value is `null` in Angular forms. I don't really want to extend it to support everything and besides `?.` is transpiled to a more verbose structure anyway.